### DISCUSSION TITLE
How to give Table BoolColumn a default (not const) value?

### DISCUSSION THREAD

**Discussion by archsolar** [simolus3/drift#2728] [Q&A] [answered]
Hi, 
```
class Profiles extends Table {
...
BoolColumn get archived => boolean()();
...
}
```

How do I give this a default `false `value. Like `Value(false)`?

Not a const one like this:
`boolean().withDefault(const Constant(false))();
`
Or am I supposed to make it const and then replace it when I want to edit the value?

**Comment by simolus3** [comment, marked as answer]
If you want the default value to be stored in the database as part of the table's definition, `Constant(false)` would be the appropriate value. If you want to change the value later, you'd re-run the build and write a [schema migration](https://drift.simonbinder.eu/docs/migrations/api/#changing-column-constraints) to change the default value as stored in the database.

If you anticipate that you need to change the default value frequently, an alternative would be [clientDefault](https://drift.simonbinder.eu/api/drift/buildcolumn/clientdefault). That value is filled in by drift when an insert happens and is not stored in the database itself, so you can change it without a schema migration.
